components/script/dom/urlhelper.rs

```diff
@@ -34,6 +34,59 @@ impl UrlHelper {
         })
     }
 
+    pub fn Protocol(url: &Url) -> USVString {
+        // https://url.spec.whatwg.org/#dom-urlutils-protocol
+        // FIXME: Url null check is skipped for now
+        USVString(format!("{}:", url.scheme))
+    }
+
+    pub fn Username(url: &Url) -> USVString {
+        // https://url.spec.whatwg.org/#dom-urlutils-username
+        // FIXME: Url null check is skipped for now
+        USVString(match url.username() {
+            None => "".to_owned(),
+            Some(username) => username.to_owned()
+        })
+    }
+
+    pub fn Password(url: &Url) -> USVString {
+        // https://url.spec.whatwg.org/#dom-urlutils-password
+        USVString(match url.password() {
+            None => "".to_owned(),
+            Some(password) => password.to_owned()
+        })
+    }
+
+    pub fn Host(url: &Url) -> USVString {
+        // https://url.spec.whatwg.org/#dom-urlutils-host
+        // FIXME: Url null check is skipped for now
+        let host = match url.host() {
+            None => "".to_owned(),
+            Some(host) => format!("{}", host)
```

Comment:
nit: we format `host` twice here, perhaps we shouldn't (instead do all formatting in the match below)